File: src/main/kotlin/codeacademy/bookingforum/app/report/Report.java
package codeacademy.bookingforum.app.report;

import codeacademy.bookingforum.app.comment.Comment;
import codeacademy.bookingforum.app.post.Post;
import codeacademy.bookingforum.app.user.auth.UserAuth;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;

@Entity
@Table(name = "Report")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Report {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    Long id;
    @Column(name = "reason")
    String reason;
    @Column(name = "description")
    String description;
    @Column(name = "report_date")
    LocalDateTime reportDate;
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "user_id")
    UserAuth user;
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "post_id")
    Post post;
    @ManyToOne(cascade = CascadeType.ALL)
    @JoinColumn(name = "comment_id")
    Comment comment;
}

File: src/main/kotlin/codeacademy/bookingforum/app/report/Report.kt
package codeacademy.bookingforum.app.report

import codeacademy.bookingforum.app.comment.Comment
import codeacademy.bookingforum.app.post.Post
import codeacademy.bookingforum.app.user.auth.UserAuth
import jakarta.persistence.*
import java.time.LocalDateTime

@Entity
@Table(name = "report")
data class Report(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val id: Long? = null,

    @Column(name = "reason")
    val reason: String? = "",

    @Column(name = "description")
    val description: String? = "",

    @Column(name = "report_date")
    val reportDate: LocalDateTime? = null,

    @ManyToOne(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JoinColumn(name = "comment_id")
    val commentId: Comment? = null,

    @ManyToOne(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id")
    val userID: UserAuth? = null,

    @ManyToOne(cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    @JoinColumn(name = "post_id")
    val postId: Post? = null,

)
